x-pack/plugins/ml/server/models/annotation_service/annotation.ts

```diff
@@ -332,6 +342,80 @@ export function annotationProvider({ asInternalUser }: IScopedClusterClient) {
     }
   }
 
+  /**
+   * Fetches the latest delayed data annotation per job.
+   * @param jobIds
+   * @param earliestMs - Timestamp for the end_timestamp range query.
+   */
+  async function getDelayedDataAnnotations({
+    jobIds,
+    earliestMs,
+  }: {
+    jobIds: string[];
+    earliestMs?: number;
+  }): Promise<Annotation[]> {
+    const params: estypes.SearchRequest = {
+      index: ML_ANNOTATIONS_INDEX_ALIAS_READ,
+      size: 0,
+      body: {
+        query: {
+          bool: {
+            ...(earliestMs
+              ? {
+                  must: [
+                    {
+                      range: {
+                        end_timestamp: {
+                          gte: earliestMs,
+                        },
+                      },
+                    },
+                  ],
+                }
```

Comment:
This COULD be added as a `filter`. Not sure if its faster or not.

Reply:
I wasn't sure about adding it as a filter, because `earliestMs` is changed during each rule execution 🤔

Reply:
> I wasn't sure about adding it as a filter, because earliestMs is changed during each rule execution

Why does that matter? If it doesn't change, query is cached, If it does change, the query still occurs.

Reply:
updated in 637cc907e853f25c261f23e774e07d4e5445544b